src/main/groovy/io/saagie/plugin/dataops/clients/SaagieClient.groovy

```diff
@@ -1003,27 +1002,31 @@ class SaagieClient {
 
     static getTemporaryFile(String url, boolean bool) {
         def tempJobDirectory = null
-
+    
+    
+        UUID uuid = UUID.randomUUID()
+        
         if(url){
-            tempJobDirectory = new File(url)
-            bool = tempJobDirectory.exists()
+            def tempJobDirectoryContainer = new File(url)
+            bool = tempJobDirectoryContainer.exists()
             if(!bool){
-                throw new GradleException("Could not find temporary directory, verify again please")
+                throw new GradleException("Could not find main temporary directory, verify again please")
```

Comment:
In this kind of exception, it could be great to also print the error directory name that was not found, it would help debugging